pkg/cmd/root.go

```diff
@@ -47,8 +47,14 @@ func init() {
 	RootCmd.PersistentFlags().BoolVar(&disableDockerIntegration,
 		"disable-docker-integration", false, "Do not attempt to list/stop/start"+
 			" docker containers which are using dvol volumes")
+}
+
+func initConfig() {
+	if _, err := os.Stat(basePath); err != nil {
+		os.MkdirAll(basePath, 0600)
+	}
 
 	viper.SetConfigName("config")
-	viper.AddConfigPath("$HOME/.dvol")
+	viper.AddConfigPath(basePath)
 	viper.ReadInConfig()
```

Comment:
I'm leaning towards reporting the parsing error if the file exists, thinking of the situation where a config file has been corrupted in some way and can no longer be parsed. As a side note however, the viper library will look for any `config` file within the config path, ignoring the extension. If you happen to have, for example, `config.json` within that directory, viper will use it instead of the yaml file because json is ahead of yaml in the list of supported extensions. We will write to the yaml file but never read from it. We could use `viper.ConfigFileUsed()` and verify that it's used the file we expect.